server/controllers/cohort.js

```diff
@@ -48,12 +49,14 @@ const getCohortsMetaData = (req, resp) => {
     .select('_id name description favIds memberIds ownerIds')
     .sort({ createdAt: -1 })
     .exec()
-    .then(docs =>
-      docs
-        ? resp.status(200).send(responseWithCohorts(docs, currentUserId))
-        : resp.status(404).send({ message: 'No cohorts data found.' })
-    )
-    .catch(err =>
+    .then(docs => {
+      if (!docs) {
+        return resp.status(400).send({ message: 'No cohorts data found.' });
```

Comment:
You should throw an error here to make this promise rejected. The return doesn't do this and controller will try to send a second response which will cause an error/warning.